### PR TITLE
*: support variable `log_bin`

### PR DESCRIPTION
**Pull request by aliiohs**
Support  log_bin

i changed  some files,like:
sessionctx/variable/sysvar.go:395 
sessionctx/variable/sysvar.go:751 
sessionctx/variable/varsutil.go:351
executor/set_test.go:223

### CODE REVIEW

File: sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -741,6 +748,8 @@ const (
 	InnodbLockWaitTimeout = "innodb_lock_wait_timeout"
 	// SQLLogBin is the name for 'sql_log_bin' system variable.
 	SQLLogBin = "sql_log_bin"
+	//LogBin is the name for 'log_bin' system variable.
```
Comment (WangXiangUSTC) [inline sessionctx/variable/sysvar.go]: please add a blank after `//`

File: sessionctx/variable/sysvar.go [hunk 2]
```diff
@@ -386,7 +393,7 @@ var defaultSysVars = []*SysVar{
 	{ScopeGlobal, "log_syslog_include_pid", ""},
 	{ScopeSession, "last_insert_id", ""},
 	{ScopeNone, "innodb_ft_cache_size", "8000000"},
-	{ScopeNone, "log_bin", "OFF"},
+	{ScopeNone, LogBin, boolToStatusStr(config.GetGlobalConfig().Binlog.Enable)},
```
Comment (WangXiangUSTC) [inline sessionctx/variable/sysvar.go]: set here seems not work, `defaultSysVars` is used in  `init` function, this function will run when this path be imported, and the Binlog.Enable may be not set at this time.

File: tidb-server/main.go [hunk 3]
```diff
@@ -439,6 +445,7 @@ func setGlobalVars() {
 
 	variable.SysVars[variable.TIDBMemQuotaQuery].Value = strconv.FormatInt(cfg.MemQuotaQuery, 10)
 	variable.SysVars["lower_case_table_names"].Value = strconv.Itoa(cfg.LowerCaseTableNames)
+	variable.SysVars["log_bin"].Value = boolToStatusStr(config.GetGlobalConfig().Binlog.Enable)
```
Comment (WangXiangUSTC) [inline tidb-server/main.go]: how about use  `variable.LogBin`

File: tidb-server/main.go [hunk 4]
```diff
@@ -418,6 +418,12 @@ func validateConfig() {
 		os.Exit(-1)
 	}
 }
+func boolToStatusStr(b bool) string {
```
Comment (WangXiangUSTC) [inline tidb-server/main.go]: same with `boolToStatusStr` in executor/set_test.go, we can only retain one

File: tidb-server/main.go [hunk 5]
```diff
@@ -24,11 +24,11 @@ import (
 	"sync/atomic"
 	"time"
 
-	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go"
 	"github.com/pingcap/errors"
 	"github.com/pingcap/parser/mysql"
 	"github.com/pingcap/parser/terror"
-	pd "github.com/pingcap/pd/client"
```
Comment (WangXiangUSTC) [inline tidb-server/main.go]: don't need modify this line and Line 27

File: sessionctx/variable/sysvar.go [hunk 6]
```diff
@@ -115,6 +115,14 @@ func boolToIntStr(b bool) string {
 	return "0"
 }
 
+// BoolToStatusStr converts bool to string,for example "ON" or "OFF".
```
Comment (WangXiangUSTC) [inline sessionctx/variable/sysvar.go]: add a blank after `,`

**Comment by WangXiangUSTC** [comment]
LGTM

**Comment by aliiohs** [comment]
@jackysp  i already added the test case for 'log_bin'

File: sessionctx/variable/varsutil.go [hunk 7]
```diff
@@ -340,6 +340,15 @@ func ValidateSetSystemVar(vars *SessionVars, name string, value string) (string,
 			return "0", nil
 		}
 		return value, ErrWrongValueForVar.GenWithStackByArgs(name, value)
+	case LogBin:
```
Comment (zimulala) [inline sessionctx/variable/varsutil.go]: Could we put it to line 335? Then we can remove line344-351.

Comment (aliiohs) [reply sessionctx/variable/varsutil.go]: LogBin need to return "ON" or "OFF", not "0" or "1"

Comment (aliiohs) [reply sessionctx/variable/varsutil.go]: so i will remove  LogBin to line360

Comment (aliiohs) [reply sessionctx/variable/varsutil.go]: so i will merge  'LogBin' and 'TiDBEnableTablePartition'

Comment (aliiohs) [reply sessionctx/variable/varsutil.go]: @zimulala so i will merge 'LogBin' and 'TiDBEnableTablePartition'

File: sessionctx/variable/varsutil.go [hunk 8]
```diff
@@ -348,7 +348,7 @@ func ValidateSetSystemVar(vars *SessionVars, name string, value string) (string,
 			return value, nil
 		}
 		return value, ErrWrongValueForVar.GenWithStackByArgs(name, value)
-	case TiDBEnableTablePartition:
+	case TiDBEnableTablePartition, LogBin:
```
[the remark this answers is not in the document]
Comment (aliiohs) [reply sessionctx/variable/varsutil.go]: so it still keep an independent case.

[the remark this answers is not in the document]
Comment (aliiohs) [reply sessionctx/variable/varsutil.go]: @zimulala  return "1" or "2", is it  expected?

[the remark this answers is not in the document]
Comment (aliiohs) [reply sessionctx/variable/varsutil.go]: @zimulala return "1" or "0", is it expected?

[the remark this answers is not in the document]
Comment (aliiohs) [reply sessionctx/variable/varsutil.go]: @zimulala  I think I changed the return value to "0" and "1" is not a good way

[the remark this answers is not in the document]
Comment (aliiohs) [reply sessionctx/variable/varsutil.go]: @winkyao  I think I changed the return value to "0" and "1" is not a good way, so i want to keep it in  independent case.

File: sessionctx/variable/varsutil.go (hunk 7 above)
Comment (zimulala) [reply sessionctx/variable/varsutil.go]: No. I mean you can merge `LogBin` with `GeneralLog`. `TiDBEnableTablePartition` has the value of `AUTO`, which is different from `LogBin`.

Comment (aliiohs) [reply sessionctx/variable/varsutil.go]: ,the return value is "0" or “1”, is it expected?

Comment (aliiohs) [reply sessionctx/variable/varsutil.go]: The return value is "0" or “1”,  is it expected? @zimulala

Comment (zimulala) [reply sessionctx/variable/varsutil.go]: I think it's OK. We can handle `LogBin` like `SQLLogBin`.  Then we can remove `BoolToStatusStr`. 
We will handle #9357 in the next PR.

File: sessionctx/variable/varsutil.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (aliiohs) [reply sessionctx/variable/varsutil.go]: ok, i will change the code later.

File: sessionctx/variable/varsutil.go (hunk 7 above)
Comment (aliiohs) [reply sessionctx/variable/varsutil.go]: ok, i will change the code later. @zimulala

File: executor/set_test.go [hunk 9]
```diff
@@ -15,6 +15,7 @@ package executor_test
 
 import (
 	"context"
+	"github.com/pingcap/tidb/config"
```
Comment (jackysp) [inline executor/set_test.go]: Keep the sys libs seperate with the 3rd libs. Please move it down to line 20.

Comment (aliiohs) [reply executor/set_test.go]: i  already finished it

**Pull request merged**